tabs/createWebauthn.tsx

```diff
@@ -0,0 +1,84 @@
+import { useCallback, useEffect, useState, type DependencyList } from "react"
+
+import { sendToContentScript } from "@plasmohq/messaging"
+
+import {
+  ContentMethod,
+  handleCreateWebauthn,
+  type ContentRequestArguments
+} from "~contents/messages"
+import * as UrlObject from "~tabs/utils/urlObject"
+
+export const CreateWebauthn = () => {
+  const [tabParams, setTabParams] = useState<Record<string, any> | undefined>()
+  const [cred, setCred] = useState<Record<string, any> | undefined>()
+
+  // When a Content Script requests window or tab creation via Service Worker Messaging, passing req.sender.tab.id to the newly created window or tab is recommended. This facilitates sending processed information back to the original Content Script using the specified tabId.
+  useEffect(() => {
+    const urlParams = window.location.href.split("?")
+    const params = UrlObject.objectFromUrlParams(
+      urlParams[urlParams.length - 1].replace(window.location.hash, "")
+    )
+    setTabParams(params)
+  }, [])
+
+  const buttonCreateWebauthnViaContentsDeps: DependencyList = [tabParams]
+  const buttonCreateWebauthnViaContents = useCallback(async () => {
+    const tabId = tabParams.tabId ? tabParams.tabId : undefined
+    const contentReq = {
+      tabId: tabId,
+      name: "",
+      body: {
+        method: ContentMethod.content_createWebauthn,
+        params: tabParams
+      } as ContentRequestArguments
+    }
+    // When requesting the Content Script to create a WebAuthn, the response is consistently undefined.
+    const contentRes = await sendToContentScript(contentReq)
+
+    console.log(
+      `[tab][createWebauthnViaContents] contentRes: ${JSON.stringify(
+        contentRes,
+        null,
+        2
+      )}`
+    )
+    setCred(contentRes)
+  }, buttonCreateWebauthnViaContentsDeps)
```

Comment:
nit: And same for `buttonCreateWebauthn`

```suggestion
  const buttonCreateWebauthnViaContents = useCallback(async () => {
    const tabId = tabParams.tabId ? tabParams.tabId : undefined
    const contentReq = {
      tabId: tabId,
      name: "",
      body: {
        method: ContentMethod.content_createWebauthn,
        params: tabParams
      } as ContentRequestArguments
    }
    // When requesting the Content Script to create a WebAuthn, the response is consistently undefined.
    const contentRes = await sendToContentScript(contentReq)

    console.log(
      `[tab][createWebauthnViaContents] contentRes: ${JSON.stringify(
        contentRes,
        null,
        2
      )}`
    )
    setCred(contentRes)
  }, [tabParams])
```

Reply:
I initially considered using a separate DependencyList for each button handler so that the associated DependencyList could be easily found without scrolling.
It probably won't provide significant benefit here, so I would consider removing it first.